api/common_interfaces.go

```diff
@@ -0,0 +1,7 @@
+package api
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type CommonResource interface {
+	MatchConditions() (*metav1.LabelSelector, string, *bool)
```

Comment:
I think we can split this further into `MatchLabels()`,`MatchNamespace()` and `AllowCrossNamespace()`. This would get rid of the `selector,_,_ := ..` at later points